### DISCUSSION TITLE
Slider Range Changing

### DISCUSSION THREAD

**Discussion by JoshStrobl** [hecrj/iced#1010] [Q&A] [answered]
How is one expected to change the slider range values after construction of the Slider, or is the current method essentially to destroy and reconstruct a new slider whenever you want the range to be changed?

Expanding on this, how are we expected to set the value of the Slider rather than solely rely on user input?

Usecase:

- In an audio manager, each track has a varying length. So if you wanted each step of 1 second and expand the range to be inclusive of the start and end of the track, you would need to be able to change the range, reasonably without having to destroy the slider widget.
- Changing the value is useful outside user input when updating the slider position based on external input, such as gstreamer progress changes, user interaction with other elements like backwards / forwards buttons, etc.

**Comment by 90-008** [comment, marked as answer]
Slider's `new` method takes a `range` and a `value` (https://github.com/hecrj/iced/blob/master/native/src/widget/slider.rs#L67), so you can do all of the things you said. Using an "external input" would basically be just storing it in your state and passing it via `value`. Same for changing the `range`.

As for "without having to recreate", all widgets are essentially objects that get created during `view` method and are consumed, so i'm not sure what you mean by that.

**Reply by manokara** [reply]
iced widgets are essentially stateless and are allocated on the stack, if allocation overhead is what he's thinking about. Like yusdacra said, you're supposed to create them every time in the `view()` method while the state data is passed from the application struct to the widget's constructor.

**Reply by JoshStrobl** [reply]
Yea, this was my line of thinking. Just coming from the GTK land where things are different and iced obviously follows a completely different model, so was trying to wrap my head around that. `view()` and reconstructing seems to be the appropriate method in performing this.
